Multisig/Features/Security/SensitiveStore.swift

```diff
@@ -0,0 +1,178 @@
+    }
+
+    func find(dataID: DataID, password: String?) throws -> EthPrivateKey? {
+        // Get encrypted signing key
+        guard let encryptedSigningKey = try store.find(KeychainItem.generic(id: dataID.id, service: dataID.protectionClass.service())) as? Data else {
+            return nil
+        }
+        // If no password given retrieve the password from store
+        let storedPassword = try store.find(KeychainItem.generic(id: SensitiveStore.derivedPasswordTag, service: ProtectionClass.sensitive.service())) as! Data?
+        let password = password != nil ? password?.data(using: .utf8) : storedPassword
+
+        // Get access to secure enclave key
+        let sensitiveKEK = try store.find(KeychainItem.enclaveKey(password: password)) as! SecKey
+
+        // Get access to encrypted sensitive key
+        let encryptedSensitiveKey = try store.find(KeychainItem.generic(id: SensitiveStore.sensitiveEncryptedPrivateKeyTag, service: ProtectionClass.sensitive.service())) as? Data
+        // Decrypt sensitiveKey
+        let decryptedSensitiveKeyData = try encryptedSensitiveKey?.decrypt(privateKey: sensitiveKEK)
+
+        // Restore sensitive key from Data
+        var error: Unmanaged<CFError>?
+        guard let decryptedSensitiveKey: SecKey = SecKeyCreateWithData(decryptedSensitiveKeyData! as CFData, try KeychainItem.ecKeyPair.creationAttributes(), &error) else {
+            // will fail here if password was wrong
+            throw error!.takeRetainedValue() as Error
+        }
+        // Decrypt signer key with sensitive key
+        return try encryptedSigningKey.decrypt(privateKey: decryptedSensitiveKey)
+    }
+
+    func changePassword(from oldPassword: String, to newPassword: String) {
+
+    }
+
+    func changeSettings() {
+
+    }
+
+    func initialize() throws {
+        // 1. create password
+        let passwordData = createRandomBytes(32)
+        // store password
+        let passItem = KeychainItem.generic(
+                id: SensitiveStore.derivedPasswordTag,
+                service: ProtectionClass.sensitive.service(),
+                data: passwordData
+        )
+        try store.create(passItem)
+
+        // 2. create kek
+        // kek security attributes are
+        // based on the user's security settings
+        // and on the defaults
+        // that is application/module-specific
+        let kekItem = KeychainItem.enclaveKey(
+                tag: SensitiveStore.sensitivePrivateKEKTag,
+                password: passwordData,
+                access: [.applicationPassword]
+        )
+        let keyEncryptionKey: SecKey = try store.create(kekItem) as! SecKey
+
+        // 3. create key pair
+        let privateKeyItem = KeychainItem.ecKeyPair // is it a pair though? also, why no tag? because it is not stored directly.
+        let sensitiveKey: SecKey = try store.create(privateKeyItem) as! SecKey
```

Comment:
if this `try` fails, then the system is in the state where `password` is created. Do we need to care about this or it will be overridden on the next initialization?

Reply:
IsInitialized() is false until the pub key of the sensitive key is stored. Last thing in this block. So if it crashes here, the password will be overwritten in the next attempt to initialize().